Create Control Line

aidangannon opened the pull request:
create sockets implementation

aidangannon committed:
created classes
- ControlLineSockets.cs
- IControlLine.cs
- SocketFactory.cs

insufficientmemory reviewed (approved):
We need to go through this.

aidangannon committed:
amended design
- refactored udp client out of control line class
- created raw socket interface

aidangannon committed:
amended design
- set up ControlLineSockets.cs to test

aidangannon committed:
Merge branch 'master' into create-control-line-sockets

aidangannon committed:
Added Unit Tests
- written test case for when data over sockets cant be sent, given send is called
- written test case for when sockets connection cannot be opened, given send is called

aidangannon committed:
Deleted Removed Projects From Solution

aidangannon committed:
refactored design
- amended control line to have just Send method
- created operation dto to encapsulate hardware operation
- created hardware exceptions

aidangannon committed:
Written Unit Tests For Send Operation Over Control Line
- When Connection Not Opened
- When Payload Not Sent
- When Response Not Retrieved
- When Control Line Times Out
- When Device Error Occurs
- When Response Retrieved Successfully

aidangannon committed:
Implemented And Amended Unit Tests For Send Operation Over Control Line

aidangannon committed:
Amended ControlLine::SendOperation()
- added param timeout period

aidangannon committed:
Amended ControlLine::SendOperation() Tests
- changed timeout period

aidangannon committed:
Refactored Threading Out Of ControlLine
- implemented as separate dependency
- mocked it

aidangannon committed:
Amended Folder Structure
- added tests folder for integration and unit

aidangannon committed:
Written Integration Tests For ThreadOperations::WaitUntilTimeout
- when call throws exception
- when call times out
- when call doesnt time out

aidangannon committed:
Written Integration Tests
- for sockets (sandwich)
- for threading (bottom up)

aidangannon committed:
Added Conf File For Integration Tests

aidangannon committed:
Modified Cooldown Time For Socket Integration Tests

aidangannon committed:
Written Integration Tests For Send Operation
- when value light value is retrieved successfully
- when axis is moved relative successfully
- when axis is moved absolute unsuccessfully

aidangannon committed:
Removed Socket Tests
- amended control line -> send operation integration tests

aidangannon committed:
Removed Socket Tests
- amended control line -> send operation integration tests, changed port

aidangannon committed:
Removed Socket Tests
- amended control line -> send operation integration tests, changed timeout period

aidangannon committed:
Removed Socket Tests
- amended control line -> send operation integration tests, changed timeout period

aidangannon committed:
Removed Socket Tests
- amended control line -> send operation integration tests, remove server timeout period

aidangannon committed:
Removed Socket Tests
- removed everything but sanity check

aidangannon committed:
renamed folder structures

aidangannon committed:
refactored unit tests, refactored ControlLineSockets::SendOperation
- added pre-conditions into send operation method
- refactored unit tests set up methods and naming convention

aidangannon committed:
refactored ControlLineSockets::SendOperation
- throws SocketException instead of custom ControlLineOfflineException ( which is vague and unclear )

aidangannon committed:
moved unit tests namespace
- moved namespace of ControlLine::SendOperation() tests

aidangannon committed:
removed error validator
- moved dependency out of ControlLineSockets.cs

aidangannon committed:
removed error validator
- moved dependency out of ControlLineSockets.cs

aidangannon committed:
removed operation timeout parameter

aidangannon committed:
added tests for valid response, added post conditions
- added post condition: if response status code is greater than 0 and less than 4
- added post condition: if response length is 8 bytes

aidangannon committed:
removed integration tests

aidangannon committed:
amended unit test
- return 8 bytes when SocketClient.Receive() is called

aidangannon committed:
removed business errors (sad path) from hardware coms logic
- removed exceptions
